src/components/Tooltip/index.js

```diff
@@ -1,36 +1,36 @@
 // @noSnapshot
-import React, { PureComponent } from 'react'
+import React, { useState } from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
 
 import { positionType } from './types'
 import { Bubble, ChildrenContainer, Container } from './styles'
 import Message from './exposedStyles/Message'
 
-class Tooltip extends PureComponent {
-  state = {
-    showTooltip: false,
-  }
-
-  onShowTooltip = () => this.setState({ showTooltip: true })
-
-  onHideTooltip = () => this.setState({ showTooltip: false })
-
-  render() {
-    const { children, message, position } = this.props
-    const { showTooltip } = this.state
-
-    return (
-      <Container onMouseLeave={this.onHideTooltip}>
-        <Bubble position={position} showTooltip={showTooltip}>
-          <Message>{message}</Message>
-        </Bubble>
-        <ChildrenContainer onMouseOver={this.onShowTooltip} onFocus={this.onShowTooltip}>
-          {children}
-        </ChildrenContainer>
-      </Container>
-    )
-  }
+const Tooltip = ({ children, message, position }) => {
+  const [show, setShow] = useState(false)
+
+  return (
+    <Container
+      onMouseLeave={() => {
+        setShow(false)
+      }}
+    >
+      <Bubble position={position} showTooltip={show}>
+        <Message>{message}</Message>
+      </Bubble>
+      <ChildrenContainer
+        onMouseOver={() => {
```

Comment:
Can you pass functions references to the show/hide tooltip functions ? Will be more optimized than using anonymous functions